### PR TITLE
UWUW compatability

### PR DESCRIPTION
**Pull request by elliottbiondo**
This PR add two features:

1. An additional flag, -U, for mcnp2cad that allows for the creation of CUBIT groups that are compatible with UWUW, i.e. in the form ```mat:X/rho:Y```. The material name (X) is chosen to be the name of the material card from which the material originates (e.g. "m1", "m2").

2. A python script was added to read MCNP input files and create UWUW material library files, with a naming scheme compatible with the -U flag. A unit test was added for this script. The script was confirmed to work with ```uwuw_preproc```, contigent on the bug fixes [here](https://github.com/svalinn/DAGMC/pull/363).

### CODE REVIEW

File: mcnp2cad.cpp [hunk 1]
```diff
@@ -153,7 +153,14 @@ class GeometryContext {
   std::string materialName( int mat, double rho ){
     std::string ret;
     std::stringstream formatter;
-    formatter << "mat_" << mat << "_rho_" << rho;
+    if(Gopt.uwuw_names){
+      char buffer [50];
+      int n;
+      n = sprintf(buffer, "%E", rho);
```
Comment (sjackso) [inline mcnp2cad.cpp]: These three lines appear to be unused, can they be removed?

**Comment by makeclean** [comment]
Im somewhat against adding a PyNE dependency to mcnp2cad which already has enough. mcnp2cad has one function which is to create CAD from mcnp native files, what you want to test is that the groups are of the right format right? So could you instead not just read the sat file and see if the group parameter has the right name set? 

The python file which takes an mcnp input deck and creates a uwuw library, I think should probably live in tools/DAGMC. This would fix the PyNE dependency problem in mcnp2cad.

File: mcnp2cad.cpp (hunk 1 above)
Comment (elliottbiondo) [inline mcnp2cad.cpp]: My reason for the ```sprintf``` was to make the string formatting more reproducible across multiple languages, i.e. Python. This isn't a huge deal, but I could imagine it saving headaches in some situation in the future.

**Comment by elliottbiondo** [comment]
> This would fix the PyNE dependency problem in mcnp2cad.

To be clear MCNP2CAD itself doesn't take on a PyNE dependency in this PR, so I do not think this creates a dependency problem. I agree that MCNP2CAD should remain a free-standing tool.

Though I know of other uses, I think the principle use of MCNP2CAD is to convert MCNP input files into a form usable by DAGMC versions of transport codes. As I see it, everything in DAGMC is moving towards UWUW conventions. UWUW requires coordination between CUBIT group names and material library names, so it makes sense for this coordination to happen within the same project.

**Comment by makeclean** [comment]
Your test does though, which is the only part of this that Im not happy with. Im happy for the uwuw conventions to go in there, I would rather a test that doesnt need PyNE. A script that produces a uwuw library from an MCNP input I do think should live in DAGMC/tools along with uwuw_preproc.

File: mcnp2cad.cpp [hunk 2]
```diff
@@ -153,7 +153,13 @@ class GeometryContext {
   std::string materialName( int mat, double rho ){
     std::string ret;
     std::stringstream formatter;
-    formatter << "mat_" << mat << "_rho_" << rho;
+    if(Gopt.uwuw_names){
+      char rho_formatted [50];
+      sprintf(rho_formatted, "%E", rho);
```
Comment (makeclean) [inline mcnp2cad.cpp]: Could you not replaced the sprintf with something like

```std::string rho_formated;```
```rho_formatted << std::fixed << std::setprecision(8) << std::scientific << rho```

More verbose, but more std::string-y

Comment (elliottbiondo) [inline mcnp2cad.cpp]: The way I have it, it is more obvious that it is the same as:

```
"{0:E}".format(number)
```

in Python.

**Comment by elliottbiondo** [comment]
> Your test does though, which is the only part of this that Im not happy with. 

The test is for the Python script, not the feature added to MCNP2CAD.

If you'd like, I will move the Python script and corresponding test to DAGMC/tools.

**Comment by elliottbiondo** [comment]
@makeclean I went ahead and removed the Python script and opened a PR in DAGMC [here](https://github.com/svalinn/DAGMC/pull/364).

**Comment by makeclean** [comment]
Smashing, I'll take a look tomorrow :)

**Comment by makeclean** [comment]
Thanks @elliottbiondo

**Pull request merged**